--- work/src/cmd/ksh93/sh/xec.c	2012-10-13 01:48:35.000000000 -0500
+++ work/src/cmd/ksh93/sh/xec.c	2012-10-13 01:48:35.000000000 -0500
@@ -1062,7 +1062,7 @@
 			io = t->tre.treio;
 			if(shp->envlist = argp = t->com.comset)
 			{
-				if(argn==0 || (np && nv_isattr(np,(BLT_DCL|BLT_SPC))))
+				if(argn==0 || (np && (nv_isattr(np,BLT_DCL)||(!command && nv_isattr(np,BLT_SPC)))))
 				{
 					Namval_t *tp=0;
 					if(argn)
